pytest/test_register.py

```diff
@@ -1751,3 +1751,168 @@ def test(x):
             if 'avgEstimatedLagMS' in d.keys():
                 env.assertContains('0', d['avgEstimatedLagMS'])
             
+@gearsTest()
+def testPauseUnpause(env):
+    script = '''
+def test(x):
+    execute('incr', 'counter{%s}' % (x['key']))
+
+regId = GB('StreamReader').foreach(test).register(mode='sync')
+GB('CommandReader').map(lambda x: regId).register(mode='sync', trigger='get_reg_id')
+    '''
+    env.expect('rg.pyexecute', script).ok()
+    verifyRegistrationIntegrity(env)
+
+    regId = env.execute_command('RG.TRIGGER', 'get_reg_id')[0]
+
+    conn = getConnectionByEnv(env)
+
+    conn.execute_command('xadd', 'x', '*', 'foo', 'bar')
+    conn.execute_command('xadd', 'y', '*', 'foo', 'bar')
+
+    counterX = conn.execute_command('get', 'counter{x}')
+    counterY = conn.execute_command('get', 'counter{y}')
+    env.assertEqual(counterX, '1')
+    env.assertEqual(counterY, '1')
+
+    env.execute_command('RG.PAUSEREGISTRATIONS', regId)
+
+    conn.execute_command('xadd', 'x', '*', 'foo', 'bar')
+    conn.execute_command('xadd', 'y', '*', 'foo', 'bar')
+
+    counterX = conn.execute_command('get', 'counter{x}')
+    counterY = conn.execute_command('get', 'counter{y}')
+    env.assertEqual(counterX, '1')
```

Comment:
Can't all of these env.assertEqual be wrtten as (example):

assert counterX == '1'

Reply:
I prefer to keep the convention we already have

Reply:
We should open another item to clean these and adopt modern asserts. Fair - out of scope.